Default Constructors/Destructors

ax3l opened the pull request:
Address clang-tidy [`hicpp-use-equals-default`](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-equals-default.html).
Use `= default` for trivial constructors and destructors.

ax3l commented:
Is it possible that `HDF5IOHandlerImpl::~HDF5IOHandlerImpl()` needs to be declared as `~HDF5IOHandlerImpl() override;` instead of `virtual ~HDF5IOHandlerImpl();`?

anokfireball commented:
Yes. Yes, it should just like all other ```IOHandlerImpl```s that inherit from ```AbstractIOHandlerImpl```.
... But you already did that. :hourglass:

ax3l commented:
Nope, did not change their declarations yet. They don't declare `override` yet.

ax3l commented on include/openPMD/IO/ADIOS/ADIOS1IOHandler.hpp:
```diff
@@ -47,7 +47,7 @@ namespace openPMD
     public:
         ADIOS1IOHandler(std::string const& path, AccessType);
-        ~ADIOS1IOHandler() override;
+        ~ADIOS1IOHandler() override = default;
```
Hm, can't move those up since `ADIOS1IOHandlerImpl` is still incomplete here.

anokfireball commented:
> Nope, did not change their declarations yet. They don't declare override yet.

My bad. You did the Handlers, not the HandlerImpls.

> ADIOS1IOHandlerImpl is still incomplete here.

Generally speaking: Why do you want to put the ```= default``` specifications inside the headers? They can be put in the ```.cpp``` sources just as well. This does not clutter the interfaces and does not enforce as many constraints.

Header ```test.hpp```
```cpp
class Test
{
public:
    Test();
    ~Test();
};
```

Source ```test.cpp```
```cpp
#include "test.hpp"

Test::Test() = default;
Test::~Test() = default;
```

ax3l committed:
Default Constructors/Destructors

Address clang-tidy `hicpp-use-equals-default`.
Use `= default` for trivial constructors and destructors.

ax3l commented:
Yeah, `coveralls`... *relative* coverage decreased because I removed LOC ;)

ax3l commented:
@C0nsultant works! :)
